src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -125,17 +131,26 @@ private void endTurn(GameBean game) throws GameException {
             Integer newPreparationCycle = game.getPreparationCycle();
             boolean preparationFinished = newPreparationCycle == null && !game.getCurrentMove().equals(1);
             shouldUpdateNextMove = (preparationFinished || previousPreparationCycle.equals(newPreparationCycle));
+        } else {
```

Comment:
reason of my comment not in readability, if-else condition reads good. I don't like that you do different actions in different places. Before you had only 2 places where state of game changed, now you have 3 places, that are baing called in different cases